hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/nssummary/NSSummaryAdmin.java

```diff
@@ -86,86 +85,47 @@ public Class<?> getParentType() {
-          "Bucket layout couldn't be verified for path: " + ofsPath +
-              ". Exception: " + e);
-      return false;
-    }
-  }
-
-  public boolean isObjectStoreBucket(String path) throws IOException {
-    OFSPath ofsPath = new OFSPath(path,
-        OzoneConfiguration.of(getOzoneConfig()));
-
+  private boolean isObjectStoreBucket(OzoneBucket bucket, ObjectStore objectStore) {
     boolean enableFileSystemPaths = getOzoneConfig()
         .getBoolean(OMConfigKeys.OZONE_OM_ENABLE_FILESYSTEM_PATHS,
             OMConfigKeys.OZONE_OM_ENABLE_FILESYSTEM_PATHS_DEFAULT);
-
-    OzoneClient ozoneClient = OzoneClientFactory.getRpcClient(getOzoneConfig());
-    ObjectStore objectStore = ozoneClient.getObjectStore();
-
     try {
-      OzoneBucket bucket = objectStore.getVolume(ofsPath.getVolumeName())
-          .getBucket(ofsPath.getBucketName());
-
       // Resolve the bucket layout in case this is a Link Bucket.
       BucketLayout resolvedBucketLayout =
           OzoneClientUtils.resolveLinkBucketLayout(bucket, objectStore,
               new HashSet<>());
-
       return resolvedBucketLayout.isObjectStore(enableFileSystemPaths);
     } catch (IOException e) {
       System.out.println(
-          "Bucket layout couldn't be verified for path: " + ofsPath +
-              ". Exception: " + e);
+          "Bucket layout couldn't be resolved. Exception thrown: " + e);
       return false;
     }
   }
 
   /**
-   * Checking if the bucket is part of the path.
+   * Checks if bucket is OBS bucket or if bucket is part of the path.
    * Return false if path is root, just a volume or invalid.
+   * Returns false if bucket is part of path but not a OBS bucket.
    * @param path
-   * @return true if the bucket
-   * is not part of the given path.
+   * @return true if bucket is OBS bucket or not part of provided path.
    * @throws IOException
    */
-  public boolean bucketIsPresentInThePath(String path) throws IOException {
+  public boolean isNotValidBucketOrOBSBucket(String path) {
     OFSPath ofsPath = new OFSPath(path,
         OzoneConfiguration.of(getOzoneConfig()));
-
-    OzoneClient ozoneClient = OzoneClientFactory.getRpcClient(getOzoneConfig());
-    ObjectStore objectStore = ozoneClient.getObjectStore();
-
-    try {
-      OzoneBucket bucket = objectStore.getVolume(ofsPath.getVolumeName())
+    OzoneBucket bucket;
+    try (OzoneClient ozoneClient = OzoneClientFactory.getRpcClient(getOzoneConfig())) {
+      ObjectStore objectStore = ozoneClient.getObjectStore();
+      // Checks if the bucket is part of the path.
+      bucket = objectStore.getVolume(ofsPath.getVolumeName())
           .getBucket(ofsPath.getBucketName());
-
-      return Objects.nonNull(bucket);
+      return isObjectStoreBucket(bucket, objectStore);
     } catch (IOException e) {
       System.out.println(
           "Bucket layout couldn't be verified for path: " + ofsPath +
               ". Exception: " + e);
-      return false;
     }
+    return false;
```

Comment:
`true` should be returned if bucket is not found.

```
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 20.55 s <<< FAILURE! -- in org.apache.hadoop.ozone.shell.TestNSSummaryAdmin
org.apache.hadoop.ozone.shell.TestNSSummaryAdmin.testNSSummaryCLIRoot -- Time elapsed: 0.344 s <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertTrue.failNotTrue(AssertTrue.java:63)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:36)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:31)
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:183)
	at org.apache.hadoop.ozone.shell.TestNSSummaryAdmin.testNSSummaryCLIRoot(TestNSSummaryAdmin.java:119)
```

https://github.com/devmadhuu/ozone/actions/runs/7579739079/job/20644851580#step:6:141